### PR TITLE
feat: allow dashbaord renderer to optionally auto refresh queries

### PR DESCRIPTION
**Pull request by filipgutica**
# Summary

https://konghq.atlassian.net/browse/MA-2748

https://www.loom.com/share/e6093efdbbc6426daea4a1f33d3801e2

#### Resources

- [Creating a package docs](https://github.com/Kong/public-ui-components/blob/main/docs/creating-a-package.md)

### CODE REVIEW

File: packages/analytics/dashboard-renderer/sandbox/pages/RendererDemo.vue [hunk 1]
```diff
@@ -43,6 +43,7 @@ const context: DashboardRendererContext = {
     type: 'relative',
     time_range: '24h',
   },
+  refreshInterval: 5000,
```
Comment (mihai-peteu) [inline packages/analytics/dashboard-renderer/sandbox/pages/RendererDemo.vue:46]: i know this is just the demo page, but shouldn't we make it a little longer, like 20 seconds or so? :)

File: packages/analytics/dashboard-renderer/src/types/dashboard-renderer-types.ts [hunk 2]
```diff
@@ -6,7 +6,8 @@ import type { ExploreFilter, ExploreQuery, TimeRangeV4 } from '@kong-ui-public/a
 export interface DashboardRendererContext {
   filters: ExploreFilter[]
   timeSpec: TimeRangeV4
-  tz?: string
+  tz?: string,
+  refreshInterval?: number
```
Comment (adorack) [inline packages/analytics/dashboard-renderer/src/types/dashboard-renderer-types.ts:10]: I don't feel like this is a good fit for context.  It seems like it should be part of the definition.  Maybe we can discuss after all the meetings?

**Review by adorack — CHANGES_REQUESTED** [review]

File: packages/analytics/dashboard-renderer/src/types/dashboard-renderer-types.ts (hunk 2 above)
Comment (filipgutica) [reply packages/analytics/dashboard-renderer/src/types/dashboard-renderer-types.ts:10]: I just wanted to put it in a spot that made the least amount of code changes, and not add additional props to the component 

I thought the context is a perfect spot for it, makes sense in my mind as it applies to the whole dashboard.

Or are you thinking we should specify refresh interval per tile?

File: packages/analytics/dashboard-renderer/sandbox/pages/RendererDemo.vue (hunk 1 above)
Comment (filipgutica) [reply packages/analytics/dashboard-renderer/sandbox/pages/RendererDemo.vue]: It's just the sandbox, I wanna see it quick, not sit around for 20-30 seconds 😄

File: packages/analytics/dashboard-renderer/src/components/QueryDataProvider.vue [hunk 3]
```diff
@@ -96,7 +96,7 @@ const { data: v4Data, error, isValidating } = useSWRV(queryKey, async () => {
     emit('queryComplete')
   }
 }, {
-  refreshInterval: 0,
+  refreshInterval: props.context.refreshInterval ?? 0,
```
Comment (adorack) [inline packages/analytics/dashboard-renderer/src/components/QueryDataProvider.vue:99]: Let's set the default to 30 seconds, based on the argument that we usually want it in order to showcase real-time data.

File: packages/analytics/dashboard-renderer/src/types/dashboard-renderer-types.ts (hunk 2 above)
Comment (adorack) [reply packages/analytics/dashboard-renderer/src/types/dashboard-renderer-types.ts:10]: (Resolved out of band -- I retract my objection, sorry about that.)

**Review by adorack — CHANGES_REQUESTED** [review]

**Review by adorack — APPROVED** [review]

**Pull request merged**